### PR TITLE
[stable10] Add user additional info field for share autocomplete 

### PR DESCRIPTION
**Pull request by PVince81**
Backport of https://github.com/owncloud/core/pull/29334 to stable10

### CODE REVIEW

File: apps/files_sharing/lib/API/Share20OCS.php [hunk 1]
```diff
@@ -65,6 +65,11 @@ class Share20OCS {
 	private $config;
 
 	/**
+	 * @var string
+	 */
+	protected $additionalInfoField;
```
[the remark this answers is not in the document]
Comment (PVince81) [reply apps/files_sharing/lib/API/Share20OCS.php]: I based this on the other attributes which are protected too, don't know why either

[the remark this answers is not in the document]
Comment (jvillafanez) [reply apps/files_sharing/lib/API/Share20OCS.php]: The `ShareesController` has the attributes protected (I'll have to look for opinions on this matter, but I don't think it's a good idea anyway), so I'm fine to go along with it. However this class has the attributes private.

[the remark this answers is not in the document]
Comment (PVince81) [reply apps/files_sharing/lib/API/Share20OCS.php]: ah ok... I copy-pasted this from there

**Comment by PVince81** [comment]
@jvillafanez 6884c15607dfa1a674d598a2b6f137d1fd22aba6

File: apps/files_sharing/lib/API/Share20OCS.php [hunk 2]
```diff
@@ -97,6 +102,22 @@ public function __construct(
...
+		$this->additionalInfoField = $this->config->getAppValue('core', 'user_additional_info_field', '');
+	}
+
+	/**
+	 * Returns the additional info to display behind the display name as configured.
+	 *
+	 * @param IUser $user user for which to retrieve the additional info
+	 * @return string|null additional info or null if none to be displayed
+	 */
+	protected function getAdditionalUserInfo(IUser $user) {
```
Comment (jvillafanez) [inline apps/files_sharing/lib/API/Share20OCS.php]: protected or private?

**Comment by jvillafanez** [comment]
Just a reminder: master should be adjusted too with the new changes

**Comment by PVince81** [comment]
@jvillafanez adjusted, thanks for spotting.

- [ ] send PR to master for the private fix

**Review by jvillafanez — APPROVED** [review]

**Comment by PVince81** [comment]
master PR https://github.com/owncloud/core/pull/29493

**Pull request merged**